src/lib/constants/resourceTypes.ts

```diff
@@ -0,0 +1,5 @@
+export default {
+  STORY_LISTING: 'node--story_listing',
+  STORY: 'node--news_story',
+  QA: 'node--q_a',
+} as const
```

Comment:
I suppose we _could_ use `typeof slug === undefined` but you're absolutely right that's a bit unnatural and verbose. This was a complete brain fart on my part caused by checking `typeof slug === 'string'` below.

To the question about when it's undefined, yeah, it's for the homepage, I believe. I didn't initially have this check and then caught it in manual testing when it came through as undefined for the homepage, IIRC.

Reply:
Your thought here resonates. It felt natural to have something in `[[...slug]].tsx`, and moving it out left it feeling like we removed something that specifically applied to that file, namely, a data point that indicates the types of resources being built by that template.

But I think, for a couple reasons, that _this_ constant belongs where it now is, in `constants/resourceTypes.ts`:
1. This object is required in multiple places now. It's currently used in three files (`[[...slug]].tsx` plus two others).
2. This object is a mapping for _all_ possible resource types. We'll need something like this throughout the application, likely. It's a _different_ data object than what was previously in `[[...slug]].tsx`. It occurred to me at some point that we _might_ want to still have something in `[[...slug]].tsx` as well that indicates all the resource types that are built by that template, but since my changes ultimately split up the fetches of those resource types into individual calls, it wasn't strictly necessary (at least yet). To paint the picture more clearly, it might be useful (at some point) to have something like this in the `[[...slug]].tsx` file:
```
// [[...slug]].tsx
import RESOURCE_TYPES from '@/lib/constants/resourceTypes'
const TEMPLATE_RESOURCE_TYPES = [
  RESOURCE_TYPES.STORY_LISTING,
  RESOURCE_TYPES.NEWS.STORY,
  RESOURCE_TYPES.QA
]
```

That might be even more clear if we think that at some point we might have another file that builds a specific resource type. In fact, we toyed with this even for this PR, initially. We had the `[facility]/stories/[[...page]]` path. If that file were going to build, say, all of our story-listing pages, then we would have the following:
```
// [[...slug]].tsx
import RESOURCE_TYPES from '@/lib/constants/resourceTypes'
const TEMPLATE_RESOURCE_TYPES = [
  RESOURCE_TYPES.NEWS.STORY,
  RESOURCE_TYPES.QA
]

```
```
//[facility]/stories/[[...page]].tsx
import RESOURCE_TYPES from '@/lib/constants/resourceTypes'
const TEMPLATE_RESOURCE_TYPES = [
  RESOURCE_TYPES.STORY_LISTING,
]
```

Reply:
Oh, I certainly agree with you that it belongs here now. It just feels slightly less clear to me with it removed from slug, but it does clear things up overall for re-use... and I'll get used to it 😁